src/sleep/sleep.c

```diff
@@ -110,9 +267,14 @@ static int execute(char **modes, char **states) {
                 return log_error_errno(errno, "Failed to open /sys/power/state: %m");
 
         /* Configure the hibernation mode */
-        r = write_mode(modes);
-        if (r < 0)
-                return r;
+        if (!strv_isempty(modes)) {
+                r = write_disk_offset();
```

Comment:
so, currently if something fails in write_disk_offset() we don't log at all above debug logging. The idea is that either the functions you call log something actionable themselves, or the caller does it. Now you downgraded all logging within the function to LOG_DEBUG (or none at all), but that means you really should log here when calling it, i.e. change this:

```c
r = write_disk_offset();
if (r < 0)
        log_error_errno(r, "Failed to write hibernation disk offset: %m");
```

The rule is that every error should be logged exactly once above LOG_DEBUG level, and that means both *at least* once, and *at most* once...

Reply:
(and yupp, some of our current codebase doesn't get right, but for new code we try to strictly follow this)

Reply:
OK.  I think i'll do a second commit to move the logging in `write_mode` to be match this behavior as well then.